lib/puppet/provider/iis_virtual_directory/webadministration.rb

```diff
@@ -40,25 +41,32 @@ def create
   end
 
   def update
+    if (@resource[:ensure] == :absent)
+      # update() is called after destroy() by flush() in ../iis_powershell.rb
+      # Updating when ensure equals absent is ... illogical.
+      Puppet.debug "Skipping update after destroy for #{@resource[:name]}"
+      return
+    end
+
     Puppet.debug "Updating #{@resource[:name]}"
 
-    if @resource[:physicalpath] and ! File.exists?(@resource[:physicalpath])
-      fail "physicalpath doesn't exist: #{@resource[:physicalpath]}"
-    end
+    verify_physicalpath
 
     cmd = []
 
-    cmd << "Set-ItemProperty -Path 'IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}' -Name 'site' -Value '#{@resource[:sitename]}';" if @resource[:sitename]
-    cmd << "Set-ItemProperty -Path 'IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}' -Name 'physicalpath' -Value '#{@resource[:physicalpath]}';" if @resource[:physicalpath]
-    cmd << "Set-ItemProperty -Path 'IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}' -Name 'application' -Value '#{@resource[:application]}';" if @resource[:application]
+    # Set-ItemProperty fails when setting site or application:
```

Comment:
Does this mean the name of the site or the application name cannot be changed after creation? If so, we need change them from parameters to properties

Reply:
Yes, site and application cannot be modified with Set-ItemProperty. MODULES-6061 documents this issue. I decided to limit the scope of this PR by commenting out those commands so they do not interfere with the command that can modify physicalpath.